updates after trying slack (#5612)
RasaHQ/rasa: docs/user-guide/connectors/slack.rst

  1. To create the app go to https://api.slack.com/apps and click
     on **Create New App**.

  2. Activate the following features:

>     - `Interactive Components <https://api.slack.com/reference/block-kit/interactive-components>`_ (if your bot uses any, e.g. buttons)

Reviewer: ```suggestion
    - Interactivity & Shortcuts > Interactivity (if your bot uses any `interactive components <https://api.slack.com/reference/block-kit/interactive-components>`_ , e.g. buttons)
```

>     - Event subscriptions > Subscribe to bot events: chat.write, message.channels, message.groups, message.im & message.mpim

Reviewer: why is `chat.write` necessary?

Author: It should be removed.

>     - App Home
>       - Always Show My Bot as Online

Reviewer: ```suggestion
    - App Home > Always Show My Bot as Online
```

  3. Get the ``Bot User OAuth Access Token`` from the OAuth & Permissions page. Click ``Install App to Workspace``
     and allow access to your workspace. You will need
     to provide this value in your credentials later in these instructions. It should start
     with ``xoxb``.

>   4. The endpoint for receiving slack messages is ``http://localhost:5005/webhooks/slack/webhook``, replacing

Reviewer: ```suggestion
  4. The endpoint for receiving Slack messages is ``http://<host>:<port>/webhooks/slack/webhook``, replacing
```

>      the host and port with the appropriate values. This is the URL you should add in the "OAuth & Permissions" 

Reviewer: ```suggestion
     the host and port with the appropriate values that point to your Rasa X or Rasa Open Source deployment. This is the URL you should add in the "OAuth & Permissions" 
```

>      "Redirect URLs" and "Interactivity & Shortcuts"

Reviewer: Should we make this step 2 so that we can tell them to add the URL directly when we turn on the features?

Reviewer: Also, redirect URL should not be necessary

Author: It is required and if you delete it, the slack UI adds it back.

Author: Good idea about step 2. Cleaned up that up.

   slack:
     slack_token: "xoxb-286425452756-safjasdf7sl38KLls"

>      slack_channel: "C011GR5D33F"

Reviewer: did #channel not work?

Author: I could not get it to work

Author: It will not respond to messages in the channel if you use the name. The [connector](https://github.com/RasaHQ/rasa/blob/a095714f5e59c393d91ffa45b6eda6ac620a27a6/rasa/core/channels/slack.py#L391) won't accept it based on the name. Slack is passing us the id.

```
Received message on unsupported channel: C011GR5D33Q
```

Reviewer: Okay. I think a lot has changed since Will added features to this connector so I'm not so familiar with it anymore. Seems you know what's up though 👍
